extensions/amp-sticky-ad/0.1/amp-sticky-ad.js

```diff
@@ -138,19 +141,68 @@ class AmpStickyAd extends AMP.BaseElement {
         const borderBottom = this.element./*OK*/offsetHeight;
         this.viewport_.updatePaddingBottom(borderBottom);
         this.addCloseButton_();
-        timerFor(this.win).delay(() => {
-          // Unfortunately we don't really have a good way to measure how long it
-          // takes to load an ad, so we'll just pretend it takes 1 second for
-          // now.
-          this.vsync_.mutate(() => {
-            this.element.classList.add('amp-sticky-ad-loaded');
-          });
-        }, 1000);
+        this.scheduleLayoutForAd_();
       });
     }
   }
 
   /**
+   * Function that check if ad has been built
+   * If not, wait for the amp:built event
+   * otherwise schedule layout for ad.
+   * @private
+   */
+  scheduleLayoutForAd_() {
+    if (this.ad_.isBuilt()) {
+      this.layoutAd_();
+    } else {
+      listenOnce(this.ad_, 'amp:built', () => {
+        this.layoutAd_();
+      });
+    }
+  }
+
+  /**
+   * Layout ad, and change sticky-ad container style
+   * @private
+   */
+  layoutAd_() {
+    this.updateInViewport(dev().assertElement(this.ad_), true);
+    this.scheduleLayout(dev().assertElement(this.ad_));
+    this.delayAdLoad_();
+  }
+
+  /**
+   * Change sticky-ad container style to ad-loaded-style after certain delay.
+   * For ad type that support return render-start wait until ad layoutCallback
+   * resolve and receive amp:load:end.
+   * For ad type that don't support render-start, wait for 1 sec.
+   * @private
+   */
+  delayAdLoad_() {
```

Comment:
since this is only called by `layoutAd_`, we can merge it into `layoutAd_`. Both methods are very short.